src/views/Log.vue

```diff
@@ -14,96 +14,214 @@ GNU General Public License for more details.
+        style="white-space: pre"
+      >
+        <v-chip
+          data-cy="connected-icon"
+          variant="outlined"
+          class="flex-shrink-0"
+          v-bind="results.connected ? {
+            color: 'green',
+            prependIcon: $options.icons.mdiPowerPlug,
+          } : {
+            color: 'red',
+            prependIcon: $options.icons.mdiPowerPlugOff,
+            onClick: updateQuery
+          }"
+        >
+          {{ results.connected ? 'Connected' : 'Reconnect' }}
+        </v-chip>
+        <span
+          data-cy="log-path"
+          style="padding-left: 0.5em; color: rgb(150,150,150);"
+        >
+          {{ results.path }}
+        </span>
+      </v-col>
+    </v-row>
+
+    <!-- the log file viewer -->
+    <v-row>
+      <v-col>
+        <!-- TODO: replace v-progress-linear with v-skeleton-loader
+        https://github.com/cylc/cylc-ui/issues/1272 -->
+        <!-- <v-skeleton-loader
+          v-if="id && file && !results.path"
+          width="50%"
+          type="list-item-three-line"
+        /> -->
+        <v-progress-linear
+          v-if="id && file && !results.path"
+          indeterminate
+        />
+        <log-component
+          v-else
+          data-cy="log-viewer"
+          :logs="results.lines"
+          :timestamps="timestamps"
+        />
+      </v-col>
+    </v-row>
+  </v-container>
 </template>
 
 <script>
-import { mdiFileDocumentMultipleOutline } from '@mdi/js'
+import {
+  mdiClockOutline,
+  mdiFileDocumentMultipleOutline,
+  mdiFolderRefresh,
+  mdiPowerPlugOff,
+  mdiPowerPlug
+} from '@mdi/js'
 import { getPageTitle } from '@/utils/index'
 import graphqlMixin from '@/mixins/graphql'
 import subscriptionComponentMixin from '@/mixins/subscriptionComponent'
 import LogComponent from '@/components/cylc/log/Log.vue'
 import SubscriptionQuery from '@/model/SubscriptionQuery.model'
-import { LOGS_SUBSCRIPTION } from '@/graphql/queries'
 import { Tokens } from '@/utils/uid'
+import gql from 'graphql-tag'
+import ViewToolbar from '@/components/cylc/ViewToolbar.vue'
+import debounce from 'lodash/debounce'
 
-class LogsCallback {
-  constructor (lines) {
-    this.lines = lines
+/**
+ * Query used to retrieve data for the Log view.
+ *
+ * @type {DocumentNode}
+*/
+const LOGS_SUBSCRIPTION = gql`
+subscription LogData ($id: ID!, $file: String!) {
+  logs (id: $id, file: $file) {
+    lines
+    connected
+    path
   }
+}
+`
+
+//    error
+//    path
+//    connected
 
-  tearDown (store, errors) {
+/**
+ * Query used to retrieve available log files for the Log view.
+ *
+ * @type {DocumentNode}
+*/
+const LOG_FILE_QUERY = gql`
+query LogFiles($id: ID!) {
+  logFiles(id: $id) {
+    files
   }
+}
+`
 
-  onAdded (added, store, errors) {
-    this.lines.push(...added.lines)
+// The preferred file to start with as a list of patterns
+// The first pattern with a matching file name will be choosen
+const LOG_FILE_DEFAULTS = [
```

Comment:
Putting aside the issue of users feeding more or less anything to stderr and stdout - I would have thought that if a `job.err` has been created this would be the user's priority?

Reply:
Totally.